### DISCUSSION TITLE
Tesla Server has begun rejecting all requests

### DISCUSSION THREAD

**Discussion by xenomorpheus** [tdorssers/TeslaPy#157] [General]
In the last few days the Tesla API server has started rejecting all my requests. It says I must register my 'App', but I don't have an 'App', instead a python script.  Does anyone know the solution?  To get a valid Tesla API key I need to register my Company, my App's URL, Path etc.  I have no idea what values to give. I have tried a few URLs but the API application was rejected. 

Any assistance would be appreciated.

This is the error message.

`ERROR: get_tesla_client or vehicle_list threw exception. Retrying.
[Fri, 26 Jan 2024 15:47:56] ERROR [tesla_charger_alphaess.py.<module>:306] 412 Client Error: Endpoint is only available on fleetapi. Visit https://developer.tesla.com/docs for more info for url: https://owner-api.teslamotors.com/api/1/vehicles`

**Comment by lcailler** [comment]
Since a few days, I get an error in the api calls. Identification is successfull, but no more response with the request vehicle_list() api. So I try successfully the below solution.
I replace in the below code "VEHICLE_LIST" to "PRODUCT_LIST" and it works perfect.
Hope this will help.


### original source code :
    def vehicle_list(self):
        """ Returns a list of `Vehicle` objects """
        return [Vehicle(v, self) for v in self.api(VEHICLE_LIST)['response']]
### updated source code :
    def vehicle_list(self):
        """ Returns a list of `Vehicle` objects """
        return [Vehicle(v, self) for v in self.api(PRODUCT_LIST)['response'] if 'vehicle_id' in v.keys()]

**Reply by xenomorpheus** [reply]
Wow!  That really works.  Huge thank you.

If other people have the same error then perhaps the code may be updated?

Are you willing to create a pull request and get credit for your fix?  If not, I am happy to do a pull request.

**Reply by lcailler** [reply]
please, do a pull request

**Comment by dammad** [comment]
I have the same issue. Time for 2.8.1?

**Reply by tdorssers** [reply]
Published new version including vehicle endpoint fix.

**Reply by Invzblio** [reply]
Thanks for the release. Just tried it but it seems "drive_state" is now gone. Am I doing something wrong or it is now accessible from somehere else?

**Reply by spinkb** [reply]
ditto.  drive_state is missing for me too.  Don't see it in the cli.py results either or anything similar...specifically the GPS location.  I need to know if the car is at home if I am going to manipulate charging scenarios based on my solar output.  Otherwise is it at a supercharger and my car keeps stopping it charge because my home system says the clouds are covering my panels. :)
